lib/improver/utilities/solar.py

```diff
@@ -0,0 +1,259 @@
+        day_of_year (int):
+            Day of the year 0 to 365, 0 = 1st January
+        utc_hour (float):
+            Hour of the day in UTC in hours
+
+    Returns:
+        solar_elevation (numpy.array):
+            Solar elevation in degrees
+    """
+    declination = solar_declination(day_of_year)
+    decl = np.radians(declination)
+    hour_angle = solar_hour_angle(longitudes, day_of_year, utc_hour)
+    rad_hours = np.radians(hour_angle)
+    lats = np.radians(latitudes)
+    # Calculate solar position:
+
+    solar_elevation = (np.arcsin(np.sin(decl) * np.sin(lats) +
+                                 np.cos(decl) * np.cos(lats) *
+                                 np.cos(rad_hours)))
+
+    solar_elevation = np.degrees(solar_elevation)
+
+    return solar_elevation
+
+
+def daynight_terminator(longitudes, day_of_year, utc_hour):
+    """
+    Calculate the Latitude values of the daynight terminator
+
+    Args:
+        longitudes (numpy.array):
+            Array of longitudes
+        day_of_year (int):
+            Day of the year
+        utc_hour (float):
+            Hour of the day in UTC
+
+    Returns:
+        latitudes (numpy.array):
+            latitudes of the daynight terminator
+    """
+    declination = solar_declination(day_of_year)
+    decl = np.radians(declination)
+    hour_angle = solar_hour_angle(longitudes, day_of_year, utc_hour)
+    rad_hour = np.radians(hour_angle)
+    lats = np.arctan(-np.cos(rad_hour)/np.tan(decl))
+    lats = np.degrees(lats)
+    return lats
+
+
+class DayNightMask(object):
+    """
+    Generate a daynight mask for the provided cube
+    """
+    def __init__(self):
+        """ Initial the DayNightMask Object """
+        self.NIGHT = 0
+        self.DAY = 1
+
+    def __repr__(self):
+        """Represent the configured plugin instance as a string."""
+        result = ('<DayNightMask : '
+                  'Day = {}, Night = {}>'.format(self.DAY, self.NIGHT))
+        return result
+
+    def _create_daynight_mask(self, cube):
+        """
+        Create blank daynight mask cube
+
+        Args:
+            cube (iris.cube.Cube):
+                cube with the times and coordinates required for mask
+
+        Returns:
+            daynight_mask (iris.cube.Cube):
+                Blank daynight mask cube. The resulting cube will be the
+                same shape as the time, y, and x coordinate, other coordinates
+                will be ignored although they might appear as attributes
+                on the cube as it is extracted from the first slice.
+        """
+        daynight_mask = next(cube.slices([cube.coord('time'),
+                                          cube.coord(axis='y'),
+                                          cube.coord(axis='x')])).copy()
+        daynight_mask.long_name = 'day_night_mask'
+        daynight_mask.standard_name = None
+        daynight_mask.var_name = None
+        daynight_mask.units = 1
```

Comment:
Should this be `cf_units(Unit('1'))`?

Reply:
I did test this and I think that this is equivalent but I'll change it to use cf_units.